image/manifest.go

```diff
@@ -240,3 +244,32 @@ loop:
 	}
 	return nil
 }
+
+// ValidateManifestPatten validate the manifest's internal
```

Comment:
It should be done by a signal function, which I've fixed. But `descendants bool` seems unnecessary. I use `bytes.NewReader()` for the buffer, instead of re-opening file. The new reader handler can be released when it is finished, so `io.ReadCloser` may be not mandatory .